servo/connectors/kubernetes.py

```diff
@@ -2185,17 +2195,32 @@ class ShortByteSize(pydantic.ByteSize):
 
     @classmethod
     def validate(cls, v: pydantic.StrIntFloat) -> "ShortByteSize":
+        if isinstance(v, str) and v.replace('.', '', 1).isdigit():
+            v = float(v)
+
         if isinstance(v, str):
             try:
                 return super().validate(v)
             except:
                 # Append the byte suffix and retry parsing
                 return super().validate(v + "b")
-        elif isinstance(v, float):
+        elif isinstance(v, float): # TODO: shouldn't this apply to int as well?
```

Comment:
Been awhile since I've been in here but if I recall correctly the thinking was that an integer would specify an exact number of bytes whereas a float would be interpreted as being in `GiB` units. So `1` would be 1 byte and `1.0` would be 1 GiB.

Arguably I'm getting too cute with some of this shit but I think we have to think about it from configuration side looking in.